### DISCUSSION TITLE
Change headless chrome to firefox?

### DISCUSSION THREAD

**Discussion by Robot-Nachos** [Mogztter/asciidoctor-web-pdf#695] [Q&A]
Is it possible change out the headless chrome that asciidoctor-web-pdf uses for Firefox? I have a document that makes use of `-webkit-text-stroke` and `paint-order` is necessary to draw the outline behind the text, instead of in front of it. [Chrome doesn't support `paint-order` for HTML text correctly](https://github.com/mdn/browser-compat-data/issues/17951), but Firefox does. I haven't had any luck faking similar results using text shadows or similar methods.

My understanding is that once you've enabled Firefox with `PUPPETEER_PRODUCT=Firefox yarn add puppeteer` you just need to add `({ product: 'firefox' })` after `puppeteer.launch`. I tried doing that in browser.js but no luck.

**Comment by ggrossetie** [comment]
It should work... 

>  I tried doing that in browser.js but no luck.

What is the observed behavior? An error?

I think you should open a new issue so we can properly support Firefox.

**Reply by Robot-Nachos** [reply]
Yes, specifically I'm getting Puppeteer's 30 second timeout error. I'll open an issue.
